src/openrct2-ui/windows/Ride.cpp

```diff
@@ -5363,11 +5363,15 @@ static void WindowRideMeasurementsDesignSave(rct_window* w)
     if (gTrackDesignSaveMode)
     {
         auto errMessage = _trackDesign->CreateTrackDesignScenery(tds);
-        if (errMessage != STR_NONE)
+        if (!errMessage.Successful)
         {
-            ContextShowError(STR_CANT_SAVE_TRACK_DESIGN, errMessage, {});
+            ContextShowError(STR_CANT_SAVE_TRACK_DESIGN, errMessage.Message, {});
             return;
         }
+        if (errMessage.Message)
```

Comment:
This is not correct - this check will always be true. It needs to be compared against `STR_NONE`. This should preferably be added in the form of a method called `HasMessage()` that does this check.

Reply:
Ah I had missed that part of ResultWithMessage class definition